controllers/external-accounts.js

```diff
@@ -71,14 +71,20 @@ const linkExternalAccount = async (req, res) => {
       userId
     );
 
-    try {
-      const unverifiedRoleId = config.get("discordUnverifiedRoleId");
-      await discordServices.removeRoleFromUser(unverifiedRoleId, attributes.discordId, req.userData);
-    } catch (error) {
-      logger.error(`Error getting external account data: ${error}`);
-      return res.boom.internal("Role Deletion failed. Please contact admin.", {
-        message: "Role Deletion failed. Please contact admin.",
-      });
+    const unverifiedRoleId = config.get("discordUnverifiedRoleId");
+    const unverifiedRoleRemovalResponse = await removeDiscordRoleUtils.removeDiscordRoleFromUser(
+      req.userData,
+      attributes.discordId,
+      unverifiedRoleId
+    );
+
+    if (!unverifiedRoleRemovalResponse.success) {
+      return res.boom.internal(
+        `User details updated but ${unverifiedRoleRemovalResponse.message}. Please contact admin`,
```

Comment:
Why this condition?
Can't we just throw error from removeDiscordRoleFromUser which will be go to catch block and from there we can send 500 response?

Reply:
As this is a common function, by this way it will be much more robust as whoever using this will have full control for success(true/false) and message and they can use in all the ways possible. This is a much more flexible way I guess.